Test TestNetDefaultRestrictedConnectivity fails

Capture https://github.com/coreos/rkt/pull/2887#issuecomment-233443046

Some of the jenkins jobs on fedora-22/24 and debian-8 are failing the test.

```
19:02:44 === RUN   TestNetDefaultRestrictedConnectivity
19:02:51 --- FAIL: TestNetDefaultRestrictedConnectivity (6.69s)
19:02:51    rkt_tests.go:115: Running command: /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/tmp/functional/rkt --dir=/tmp/datadir-255390103 --local-config=/tmp/localdir-290079242 --system-config=/tmp/systemdir-688845793 --user-config=/tmp/userdir-616822732 --debug --insecure-options=image run --net=default-restricted --mds-register=false /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/tmp/functional/test-tmp/rkt-inspect-networking.aci
19:02:51    goroutineassistant.go:64: Error encountered - shutting down
19:02:51    goroutineassistant.go:65: Get http://172.16.28.2:49152: dial tcp 172.16.28.2:49152: getsockopt: connection refused
19:02:51        
19:02:51 panic: Failed to remove temporary data directory "/tmp/datadir-255390103": remove /tmp/datadir-255390103/pods/run/17421b97-926e-4b25-be37-4cecf1fb0bc1/stage1/rootfs/sys: device or resource busy [recovered]
19:02:51    panic: Failed to remove temporary data directory "/tmp/datadir-255390103": remove /tmp/datadir-255390103/pods/run/17421b97-926e-4b25-be37-4cecf1fb0bc1/stage1/rootfs/sys: device or resource busy
19:02:51 
19:02:51 goroutine 481 [running]:
19:02:51 panic(0xa3b480, 0xc8203954b0)
19:02:51    /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
19:02:51 testing.tRunner.func1(0xc8204126c0)
19:02:51    /usr/lib/go-1.6/src/testing/testing.go:467 +0x192
19:02:51 panic(0xa3b480, 0xc8203954b0)
19:02:51    /usr/lib/go-1.6/src/runtime/panic.go:443 +0x4e9
19:02:51 github.com/coreos/rkt/tests/testutils.(*dirDesc).cleanup(0xc8202491c0)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/testutils/ctx.go:67 +0x27c
19:02:51 github.com/coreos/rkt/tests/testutils.(*RktRunCtx).Cleanup(0xc8202492c0)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/testutils/ctx.go:173 +0x1be
19:02:51 runtime.Goexit()
19:02:51    /usr/lib/go-1.6/src/runtime/panic.go:340 +0xfb
19:02:51 testing.(*common).FailNow(0xc8204126c0)
19:02:51    /usr/lib/go-1.6/src/testing/testing.go:346 +0x36
19:02:51 testing.(*common).Fatal(0xc8204126c0, 0xc820209c28, 0x1, 0x1)
19:02:51    /usr/lib/go-1.6/src/testing/testing.go:383 +0x6f
19:02:51 github.com/coreos/rkt/tests/testutils.(*GoroutineAssistant).Wait(0xc8203e0d20)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/testutils/goroutineassistant.go:65 +0x195
19:02:51 github.com/coreos/rkt/tests.NewTestNetDefaultRestrictedConnectivity.func1.1(0xce9de0, 0x18)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/rkt_net_test.go:353 +0xbf0
19:02:51 github.com/coreos/rkt/tests.NewTestNetDefaultRestrictedConnectivity.func1(0xc8204126c0)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/rkt_net_test.go:355 +0x92
19:02:51 github.com/coreos/rkt/tests/testutils.TestFunc.Execute(0xdb47f0, 0xc8204126c0)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/testutils/test.go:31 +0x26
19:02:51 github.com/coreos/rkt/tests.TestNetDefaultRestrictedConnectivity(0xc8204126c0)
19:02:51    /home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/gopath/src/github.com/coreos/rkt/tests/rkt_net_nspawn_test.go:76 +0x3f
19:02:51 testing.tRunner(0xc8204126c0, 0x1238550)
19:02:51    /usr/lib/go-1.6/src/testing/testing.go:473 +0x98
19:02:51 created by testing.RunTests
19:02:51    /usr/lib/go-1.6/src/testing/testing.go:582 +0x892
19:02:51 exit status 2
19:02:51 FAIL   github.com/coreos/rkt/tests 1199.008s
19:02:51 tests/functional.mk:82: recipe for target '/home/admin/workspace/rkt-github-ci/os_type/debian-8/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.10.0+git/stamps/__tests_functional_mk_functional_tests.stamp' failed
```

cc @steveeJ

I am still seeing this sporadically with an output like this:
```
12:18:44 --- FAIL: TestNetDefaultRestrictedConnectivity (5.52s)
12:18:44 	rkt_tests.go:117: Running command: /home/fedora/workspace/rkt-github-ci/os_type/fedora-rawhide/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.15.0+git/tmp/functional/rkt --dir=/tmp/datadir-784815578 --local-config=/tmp/localdir-170059121 --system-config=/tmp/systemdir-097209884 --user-config=/tmp/userdir-072193739 --debug --insecure-options=image run --net=default-restricted --mds-register=false /home/fedora/workspace/rkt-github-ci/os_type/fedora-rawhide/stage1_flavor/coreos/builds/build-rkt-coreos/build-rkt-1.15.0+git/tmp/functional/test-tmp/rkt-inspect-networking.aci
12:18:44 	goroutineassistant.go:64: Error encountered - shutting down
12:18:44 	goroutineassistant.go:65: Get http://172.16.28.2:49152: dial tcp 172.16.28.2:49152: getsockopt: connection refused
```

--- a/tests/rkt_net_test.go
+++ b/tests/rkt_net_test.go
@@ -307,20 +307,17 @@ func NewTestNetDefaultRestrictedConnectivity() testutils.Test {
 		defer ctx.Cleanup()
 
 		f := func(argument string) {
-			httpPort, err := testutils.GetNextFreePort4()
-			if err != nil {
-				t.Fatalf("%v", err)
-			}
-			httpServeAddr := fmt.Sprintf("0.0.0.0:%v", httpPort)
+			httpPort := "8080"
 			iface := "eth0"
 
-			testImageArgs := []string{fmt.Sprintf("--exec=/inspect --print-ipv4=%v --serve-http=%v", iface, httpServeAddr)}
+			testImageArgs := []string{fmt.Sprintf("--exec=/inspect --print-ipv4=%v --serve-http=0.0.0.0:%v", iface, httpPort)}
 			testImage := patchTestACI("rkt-inspect-networking.aci", testImageArgs...)
 			defer os.Remove(testImage)
 
-			cmd := fmt.Sprintf("%s --debug --insecure-options=image run %s --mds-register=false %s", ctx.Cmd(), argument, testImage)
+			cmd := fmt.Sprintf("%s --insecure-options=image run %s --mds-register=false %s", ctx.Cmd(), argument, testImage)
 			child := spawnOrFail(t, cmd)
 
+			// Wait for the container to print out the IP address
 			expectedRegex := `IPv4: (\d+\.\d+\.\d+\.\d+)`
 			result, out, err := expectRegexWithOutput(child, expectedRegex)
 			if err != nil {
@@ -328,31 +325,18 @@ func NewTestNetDefaultRestrictedConnectivity() testutils.Test {
 			}
 			httpGetAddr := fmt.Sprintf("http://%v:%v", result[1], httpPort)
 
-			ga := testutils.NewGoroutineAssistant(t)
-			ga.Add(2)
-
-			// Child opens the server
-			go func() {
-				defer ga.Done()
-				ga.WaitOrFail(child)
-			}()
-
-			// Host connects to the child
-			go func() {
-				defer ga.Done()
-				expectedRegex := `serving on`
-				_, out, err := expectRegexWithOutput(child, expectedRegex)
-				if err != nil {
-					ga.Fatalf("Error: %v\nOutput: %v", err, out)
-				}
-				body, err := testutils.HTTPGet(httpGetAddr)
-				if err != nil {
-					ga.Fatalf("%v\n", err)
-				}
-				t.Logf("HTTP-Get received: %s", body)
-			}()
-
-			ga.Wait()
+			// Wait for the container to open the port
+			expectedRegex = `serving on`
+			_, out, err = expectRegexWithOutput(child, expectedRegex)
+			if err != nil {
+				t.Fatalf("Error: %v\nOutput: %v", err, out)
+			}
+			body, err := testutils.HTTPGet(httpGetAddr)
+			if err != nil {
+				t.Fatalf("%v\n", err)
+			}
+			t.Logf("HTTP-Get received: %s", body)
+			waitOrFail(t, child, 0)
 		}
 		f("--net=default-restricted")
 	})

